src/components/InputBar.jsx

```diff
@@ -0,0 +1,40 @@
+import './InputBar.css';
+
+function InputBar({
+  type = 'text',
+  label,
+  inputId = 'input',
+  name,
+  value,
+  placeholder,
+  disabled = false,
+  onChange,
+  onKeyDown,
+}) {
+  const eventProps = {
+    ...(onKeyDown && { onKeyDown }),
+    ...(onChange && { onChange }),
+  };
+
+  return (
+    <section className="InputBar">
+      {label && (
+        <label htmlFor={inputId} className="InputBar-label">
+          {label}
+        </label>
+      )}
+      <input
+        className="InputBar-input"
+        type={type}
+        id={inputId}
+        name={name}
+        value={value}
+        placeholder={placeholder}
+        disabled={disabled}
+        {...eventProps}
+      />
+    </section>
```

Comment:
P1:
아래처럼 하셔도 동일동작합니다.

```suggestion
  ...eventProps
}) {
  return (
    <section className="InputBar">
      {label && (
        <label htmlFor={inputId} className="InputBar-label">
          {label}
        </label>
      )}
      <input
        className="InputBar-input"
        type={type}
        id={inputId}
        name={name}
        value={value}
        placeholder={placeholder}
        disabled={disabled}
        {...eventProps}
      />
    </section>
```